pkg/api/v1alpha1/vspheredatacenterconfig_webhook.go

```diff
@@ -93,35 +93,35 @@ func validateImmutableFieldsVSphereCluster(new, old *VSphereDatacenterConfig) fi
 	if old.Spec.Server != new.Spec.Server {
 		allErrs = append(
 			allErrs,
-			field.Invalid(field.NewPath("spec", "server"), new.Spec.Server, "field is immutable"),
+			field.Forbidden(field.NewPath("spec", "server", new.Spec.Server), "field is immutable"),
```

Comment:
Same goes here on whether we want to remove that field if we are already specifying "server", as well as the ones below in this file.